compiler-core/src/type_.rs

```diff
@@ -1579,18 +1579,34 @@ fn make_type_vars(
         .try_collect()
 }
 
-fn custom_type_accessors<A>(
+#[allow(unused_qualifications)]
+fn custom_type_accessors<A: Clone + std::cmp::PartialEq>(
     constructors: &[RecordConstructor<A>],
     hydrator: &mut Hydrator,
     environment: &mut Environment<'_>,
 ) -> Result<Option<HashMap<String, RecordAccessor>>, Error> {
-    // Get the constructor for this custom type.
-    let args = match constructors {
-        [constructor] if !constructor.arguments.is_empty() => &constructor.arguments,
-        // If there is not exactly 1 constructor we return as we cannot
-        // build any constructors.
-        _ => return Ok(None),
-    };
+    if constructors.is_empty() {
+        return Ok(None);
+    }
+
+    let constructor_args_data = constructors
+        .iter()
+        .map(|constructor| constructor.arguments.as_slice())
+        .collect_vec();
+
+    let (first_constructor_arg, constructor_args) = constructor_args_data
+        .split_first()
+        .expect("No constructs with args");
+    let args = first_constructor_arg
+        .iter()
+        .filter(|data_type| {
+            constructor_args.iter().all(|set| {
+                set.iter()
+                    .map(|item| (item.label.clone(), item.type_.clone()))
```

Comment:
I think we can do this without cloning the label

Reply:
Switched to `.as_ref()`